### DISCUSSION TITLE
Callback query handler don't work

### DISCUSSION THREAD

**Discussion by cap-102** [eternnoir/pyTelegramBotAPI#2097] [Q&A]
**pyTelegramBotAPI 4.14.0 | Windows | Python 3.12.0**
So I wanted create some callback query handlers in my bot using this script:
`
https://pastebin.com/wZfumXXU
`
But it don't worked. When I tried to to add print statement in the start of my function it also printed nothing.

Then I tried this easy script:
`
https://pastebin.com/iZAW51jg
`
But it also don't worked for me

**Comment by Roninon** [comment]
Hi, [Caportabow](https://github.com/Caportabow)! Please, provide your script exception traceback, when its failed. Or, you can add next rows to your code, to get more info about your program:
`
import logging
logger = logging.getLogger('TeleBot').setLevel(logging.INFO)
`

**Comment by cap-102** [comment]
```
#---#
All contents of data/temp/ have been deleted.
#---#
Successfully Started!
Bot - @IAmBetterBot
#---#


2023-12-10 00:26:12,608 (__init__.py:1040 MainThread) INFO - TeleBot: "Starting your bot with username: [@IAmBetterBot]"
2023-12-10 00:26:12,609 (__init__.py:1057 MainThread) INFO - TeleBot: "Started polling."
```

This is full log. My bot don't fail or whenever, it just don't have any effect if I push the buttons

**Comment by Badiboy** [comment]
>   File "test.py", line 10, in start
>     markup = types.InlineKeyboardMarkup()
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: module 'types' has no attribute 'InlineKeyboardMarkup'

`import telebot, telebot.types as types`

Now works fine.

![image](https://github.com/eternnoir/pyTelegramBotAPI/assets/5613421/425348d5-19fc-4e5d-a1ca-d9956b3d50b2)

**Reply by cap-102** [reply]
I edited my "simple script" to this:
```
import telebot, telebot.types as types


botToken = "here is my bot token"
bot = telebot.TeleBot(botToken)


# Define the start command
@bot.message_handler(commands=['start'])
def start(message):
    markup = types.InlineKeyboardMarkup()
    button = types.InlineKeyboardButton('Click me!', callback_data='button_click')
    markup.add(button)

    bot.send_message(message.chat.id, 'Hello! This is an inline keyboard example.', reply_markup=markup)


# Define the callback function for the inline keyboard button
@bot.callback_query_handler(func=lambda call: True)
def callback_handler(call):
    if call.data == 'button_click':
        bot.answer_callback_query(call.id, text='Button clicked!')
        bot.edit_message_text(chat_id=call.message.chat.id, message_id=call.message.message_id, text='Button clicked!')


# Polling loop to keep the bot running
bot.polling(none_stop=True)
```
but it still don't work

**Comment by cap-102** [comment]
You won't trust me, but my issue was fixed with revoking api key of my bot

**Reply by hvosdt** [reply]
I spend a lot of time on same problem. Bot dosn't handle any callback, like nothing happened. I revoke token and magically all start working fine. Thanks to you fo advice!

**Reply by Krisen** [reply]
thx a lot

**Reply by DurnevVS** [reply]
same problem can u please explain detailed for me

**Reply by hvosdt** [reply]
> same problem can u please explain detailed for me

Just Revoke current token in Bot Father. After that u can copy new token and use it

**Reply by DurnevVS** [reply]
wow, it really helps, thx

**Reply by flexinov** [reply]
this helped, a hero we needed

**Reply by UlknAries** [reply]
BRO I HAVE BEEN SUFFERING WITH THIS PROBLEM FOR 5 FUCKING DAYS, THANK YOU SO MUCH, YOU ARE HERO

**Reply by rmvinno** [reply]
Thank you! This helped.

**Comment by coder2020official** [comment]
Hi, you have to answer to callback queries to avoid this issue. If you don't, eventually API will stop sending you this update, forcing you to revoke.
